cmd/dockerd/daemon.go

```diff
@@ -574,13 +587,43 @@ func newAPIServerConfig(cli *DaemonCli) (*apiserver.Config, error) {
 	return serverConfig, nil
 }
 
+// checkTLSAuthOK checks basically for an explicitly disabled TLS/TLSVerify
+// Going forward we do not want to support a scenario where dockerd listens
+//   on TCP without either TLS client auth (or an explicit opt-in to disable it)
+func checkTLSAuthOK(c *config.Config) bool {
+	if c.TLS == nil {
+		// Either TLS is enabled by default, in which case TLS verification should be enabled by default, or explicitly disabled
+		// Or TLS is disabled by default... in any of these cases, we can just take the default value as to how to proceed
+		return DefaultTLSValue
```

Comment:
is this shortcut correct? it currently is, but if it changes to `true`,  we won't check tlsverify below 🤔

Should probably be slightly safer to check value of defaulttlsvalue here?

Reply:
If tlsverify is explictitly set, then we don't want to error out/warn either.

So when `tls` is true, `tlsverify` should also be true (this is a change we discussed on the call) unless explicitly set to `false`. If `tlsverify` is set at all, `tls` must not be nil.

So yes, this should be correct, and is why I added the constant. When we change the default, we don't need to update the logic.